src/afrolabs/components/kafka.clj

```diff
@@ -317,7 +333,70 @@
        (apply ~n (rest strategy-specs#)))))
 
 
-;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defstrategy RenameTopicsForProducer
+  [& {:keys [rename-fn]}]
+  (when-not rename-fn
+    (throw (ex-info (str "strategy RenameTopicsForProducer requires the `rename-fn` to be specified. This function "
+                         "accepts a topic name for every message, and must return a modified version of the topic, "
+                         "OR nil, in which case the original topic value will be kept.")
+                    {})))
+  (reify
+    IProducerPreProduceMiddleware
+    (pre-produce-hook [_ msgs]
+      (map (fn [{:as   msg
+                 :keys [topic]}]
+             (assoc msg :topic (or (rename-fn topic)
+                                   topic)))
+           msgs))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- resolve-fn
+  "Resolves a configuration value that semantically points to an fn, to the actual fn.
+  `the-fn` might be a symbol, in which case the namespace will be recquired and the symbol resolved
+  or an actual `IFn`."
+  [the-fn]
+  (or (when the-fn
+        (cond
+          (fn? the-fn)       the-fn
+          (symbol? the-fn)   (let [the-fn-ref (var-get (requiring-resolve the-fn))]
+                                 (if-not (fn? the-fn-ref)
+                                   (throw (ex-info "`the-fn` must be an fn or a symbol that points to an fn."
+                                                   {:the-fn the-fn}))
+                                   the-fn-ref))
+          :else
+          (do (log/with-context+ {:the-fn the-fn}
+                (log/warn "Unable to resolve the-fn."))
+              nil)))
+      identity))
+
+(defstrategy UpdateConsumedRecordKey
+  [& {:keys [key-fn] :as params}]
+  (when-not key-fn
+    (throw (ex-info "`key-fn` must have a value."
+                    {:key-fn key-fn
+                     :params params})))
+  (let [key-fn' (resolve-fn key-fn)]
```

Comment:
Since you are already throwing in `resolve-fn` I think this `throw` should be moved in there with its own condition.